plugin/trino-faker/src/main/java/io/trino/plugin/faker/FakerPageSource.java

```diff
@@ -243,6 +238,107 @@ public void close()
         closed = true;
     }
 
+    private Generator sequenceGenerator(FakerColumnHandle handle, long rowOffset)
+    {
+        SequenceWriter writer = sequenceWriter(handle);
+
+        return new Generator()
+        {
+            long currentRowId = rowOffset;
+
+            @Override
+            public void accept(BlockBuilder blockBuilder)
+            {
+                writer.accept(blockBuilder, currentRowId++);
+            }
+        };
+    }
+
+    private SequenceWriter sequenceWriter(FakerColumnHandle handle)
+    {
+        Range genericRange = handle.domain().getValues().getRanges().getSpan();
+        Type type = handle.type();
+        // check every type in order defined in StandardTypes
+        if (BIGINT.equals(type)) {
+            LongRange range = LongRange.of(genericRange, 1, (long) handle.step().getSingleValue());
+            return (blockBuilder, rowId) -> BIGINT.writeLong(blockBuilder, range.at(rowId));
+        }
+        if (INTEGER.equals(type)) {
+            IntRange range = IntRange.of(genericRange, (long) handle.step().getSingleValue());
+            return (blockBuilder, rowId) -> INTEGER.writeLong(blockBuilder, range.at(rowId));
+        }
+        if (SMALLINT.equals(type)) {
+            IntRange range = IntRange.of(genericRange, Short.MIN_VALUE, Short.MAX_VALUE, (long) handle.step().getSingleValue());
+            return (blockBuilder, rowId) -> SMALLINT.writeLong(blockBuilder, range.at(rowId));
+        }
+        if (TINYINT.equals(type)) {
+            IntRange range = IntRange.of(genericRange, Byte.MIN_VALUE, Byte.MAX_VALUE, (long) handle.step().getSingleValue());
+            return (blockBuilder, rowId) -> TINYINT.writeLong(blockBuilder, range.at(rowId));
+        }
+        // boolean sequences are not supported
```

Comment:
nit: Maybe just replace those comments with one comment at the top claiming what is supported and what not